src/xGadgetron/pGadgetron/Gadgetron.py

```diff
@@ -437,6 +437,56 @@ def get_ISMRMRD_info(self, par):
     def get_info(self, par):
         return self.get_ISMRMRD_info(par)
 
+    def dot(self, other):
+        '''
+        Returns the dot product of the container data with another container
+        data or numpy array viewed as vectors.
+        other: ImageData or numpy array or scalar
+        '''
+        if not (issubclass(type(other), type(self))):
+            other = self.clone().fill(other)
+        return super(ImageData, self).dot(other)
```

Comment:
according to doc, since Python 3.0, you can write
```suggestion
        return super().dot(other)
```
which is a nicer, and in fact seems to say that we can write this only once somewhere, as opposed to for Gadgetron/STIR/Reg etc. No idea how to do that though.